src/android/MifareUltralight.java

```diff
@@ -35,6 +36,59 @@ public boolean unlockWithPin(long pin) throws Exception {
         return false;
     }
 
+    public boolean lockWithPin(
+            int pinPage,
+            int pinAckPage,
+            int protectionPage,
+            int firstPageToBeProtectedPage,
+            int firstPageToBeProtected,
+            long pin,
+            boolean protectAlsoReads,
+            int authenticationTryLimit
+    ) throws Exception {
+        byte[] pack;
+        // 1. Write the PIN
+        final byte[] pinAsByteArray = longToByteArray(pin);
+        byte[] response = mifare.writePage(page, pinAsByteArray);
+        if ((response != null) && (response.length >= 2)) {
+            pack = Arrays.copyOf(response, 2);
+        } else {
+            return false;
+        }
+        // 2. Write the acknowledgement
+        byte[] data = {pack[0], pack[1], (byte) 0, (byte) 0};
+        mifare.writePage(pinAckPage, data);
+
+        // 3. Read the protection page to be able to only modify certain bits and then write it again
+        response = mifare.readPages(protectionPage);
+        if ((response != null) && (response.length >= 16)) {
+            data = {
```

Comment:
Got a an error just here at compilation:
`error: illegal start of expression
            data = {
`
I'm not a Java developper so I don't know how to fix it actually :/
Maybe by modifying each entry on the array like data[0] = foo, data[1] = bar, etc .?

Reply:
Tried that with beta release 1.1.2.. :)

Reply:
Got the exact same issue on line 75 ! (sorry for the delay, I'm working at Disneyland on weekends so I'm too exhausted to code after my shift :'))

Reply:
I should've probably checked that too.. Fixed now. 

Take your time. I don't need this feature myself nor do I use the plugin really nowadays :)

Reply:
Here are the last errors I get: https://pastebin.com/pgX12tTf

Reply:
1.1.4 available :)